### ISSUE TITLE
Need more metrics about Mark Cache eviction.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#60989]
MarkCacheEvictedFiles

**Comment by CheSema** [comment]
I would rather say mark loading has to be reworked. I have wrote this 

This has to be improved:
- even in good-green path marks loading puts a task to the thread pool for only check if the marks in the cache
- in that case tasks in pool which _have_ data in cache are waiting other tasks which really load new marks, that eliminates all advantages of using cache
- there is no cancelation point, canceled query waits  until all its marks loading tasks are executed.

**Comment by rschu1ze** [comment]
Resolved by https://github.com/ClickHouse/ClickHouse/pull/80799

**Comment by CheSema** [comment]
> I would rather say mark loading has to be reworked. I have wrote it somewhere in issues comments, but I have not created designated issue yet.
> 
> This has to be improved:
> 
> * even in good-green path marks loading puts a task to the thread pool for only check if the marks in the cache
> * in that case tasks in pool which _have_ data in cache are waiting other tasks which really load new marks, that eliminates all advantages of using cache
> * there is no cancelation point, canceled query waits  until all its marks loading tasks are executed.

Is this also resolved here?

**Comment by rschu1ze** [comment]
> Is this also resolved here?

No. Open a new issue? It was not related to the original problem (better metrics) in the first place.
